pilot/pkg/bootstrap/server.go

```diff
@@ -770,6 +764,10 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 // initConfigController creates the config controller in the pilotConfig.
 func (s *Server) initConfigController(args *PilotArgs) error {
 	if len(args.MCPServerAddrs) > 0 || len(s.mesh.ConfigSources) > 0 {
+		s.mcpOptions = &coredatamodel.Options{
```

Comment:
I wonder if it might make more sense to create 2 separate controllers here:

- the old one, untouched
- the new one specifically for incremental service entries

It might make the logic a little easier to understand.  WDYT?

Reply:
That’s an interesting thought and it seems reasonable to do so. The only concern would be how much of that approach aligns with the pilot decomposition idea? From what I recall correctly one of the main goals behind the coredatamodel controller was to retire all other controllers so that we can centralize config ingestion through a well defined API. Having said that, having an additional controller would not necessarily mean we are going back to the old way of doing things, both controllers would still consume MCP. We would just split the logic to allocate one controller to legacyMCP and the other would be predominantly handling Incremental MCP and EDS updates for SSEs.

Reply:
Right, effectively they are 2 separate paths regardless so it doesn't really matter too much.  We can hide the 2 behind an aggregate controller, so that they look like a single controller from the rest of the system.  The 2 controller will eventually be merged but separating them now makes the logic easier to reason about.

Reply:
+1 on separating 'config MCP' from 'endpoint only' MCP.

Plenty of reasons - including security.

Reply:
And the 'endpoint only' MCPs can be incremental only, no case where we would want non-incremental endpoints.

Reply:
You're referring to them as "endpoints-only", but they're not really. They are k8s service + endpoints. So for each update received from Galley, either the service or endpoints may have been updated.

Are you saying that we don't care about the service portion for services in external clusters?  

If we ever want this to be the path for in-cluster synthetic Service Entries, we will have to worry about full vs incremental.

Reply:
I'm saying there are very different things - and using the same code for both is not ideal. 

The service portion of ServiceEntry is treated as any other config. There is no reason to mix it in the
same MCP adapter. We care about it just like we care about any other config CRD.

The load balancing / readiness / endpoint related code paths can be incremental from start, but 
they can use a separate MCP client/adapter that is optimized for this use case ( and as such simpler 
code on both sides)